frame/curve-amm/src/lib.rs

```diff
@@ -350,21 +378,76 @@ pub mod pallet {
 
 					let d1 = Self::get_d(&new_balances, ann).ok_or(Error::<T>::Math)?;
 					ensure!(d1 > d0, Error::<T>::WrongAssetAmount);
-					let _new_balances_b: Vec<T::Balance> = new_balances
-						.iter()
-						.map(|b_f| Ok(b_f.checked_mul_int(1u64).ok_or(Error::<T>::Math)?.into()))
-						.collect::<Result<Vec<T::Balance>, Error<T>>>()?;
 					let d1_b = d1.checked_mul_int(1u64).ok_or(Error::<T>::Math)?.into();
 					let mint_amount;
+					let mut fees = vec![FixedU128::zero(); n_coins];
+					// Only account for fees if we are not the first to deposit
 					if token_supply > zero {
-						// mint_amount = token_supply * (d1 - d0) / d0
+						// Deposit x + withdraw y would chargVe about same
+						// fees as a swap. Otherwise, one could exchange w/o paying fees.
+						// And this formula leads to exactly that equality
+						// fee = pool.fee * n_coins / (4 * (n_coins - 1))
```

Comment:
in case of 2 coints, we get `fee / 4 `, so fee is divided by 4, is it expected?

Reply:
it will be `fee / 2` right? fee * 2 / 4